### PR TITLE
airflow.models.xcom deprecations removed

### PR DESCRIPTION
**Pull request by dirrao**
airflow.models.xcom deprecations removed

### CODE REVIEW

**Comment by dirrao** [comment]
@potiuk / @eladkal, 
We have updated Xcoms provider tests. Provider tests are failing on earlier Airflow versions. Do we need to add airflow version condition-based checks? WDYT?

**Comment by potiuk** [comment]
Yes. We are running providers with previous Airflow versions - so ideally we should have them succeeded with older airflows as well. Likely a test_compat could be added to make xcom_push test work on both - see the contributing_docs for back-compatibility tests and `tests/test_utils/compat.py`

File: airflow/models/xcom.py [hunk 1]
```diff
@@ -163,67 +154,14 @@ def set(
...
-        """
         from airflow.models.dagrun import DagRun
 
-        if not exactly_one(execution_date is not None, run_id is not None):
-            raise ValueError(
-                f"Exactly one of run_id or execution_date must be passed. "
-                f"Passed execution_date={execution_date}, run_id={run_id}"
-            )
+        if not run_id:
+            raise ValueError(f"run_id must be passed. Passed run_id={run_id}")
```
Comment (uranusjr) [inline airflow/models/xcom.py:160]: Why not just remove `= None` from the argument to make it required?

File: airflow/models/xcom.py [hunk 2]
```diff
@@ -603,62 +427,15 @@ def clear(
...
-        if execution_date is not None:
-            message = "Passing 'execution_date' to 'XCom.clear()' is deprecated. Use 'run_id' instead."
-            warnings.warn(message, RemovedInAirflow3Warning, stacklevel=3)
-            run_id = (
-                session.query(DagRun.run_id)
-                .filter(DagRun.dag_id == dag_id, DagRun.execution_date == execution_date)
-                .scalar()
-            )
+        if not run_id:
+            raise ValueError(f"run_id must be passed. Passed run_id={run_id}")
```
Comment (uranusjr) [inline airflow/models/xcom.py:435]: Same here

**Comment by dirrao** [comment]
@potiuk  / @uranusjr 
Provider backward compatibility test is failing. Ideally, it should pass. Something nontrivial to me. Can someone help me with this?

**Comment by potiuk** [comment]
Looks like 'mock_context` in conftest.py should be modified to accept xcom_push with `execution_date` when Airflow < 3.0. Likely could be done by adding `**kwargs` to the xcom_push method and checking if `execution_date` is not passed when Airflow 3 (but passed when Airflow < 3)

**Comment by dirrao** [comment]
> Looks like `mock_context` in conftest.py should be modified to accept xcom_push with `execution_date` when Airflow < 3.0. Likely could be done by adding `**kwargs` to the xcom_push method and checking if `execution_date` is not passed when Airflow 3 (but passed when Airflow < 3)

Finally all the tests are passing except one unrelated test. Looks good to me.

**Review by potiuk — APPROVED** [review]

**Pull request merged**